### PR TITLE
Add support for easy operations with xportr options - `xportr_options`

### LINKED ISSUE #130: Feature Request: Provide a way to easily view xportr options
Is there a way we could allow users to easily view the options available in the package? 

image

Like a `xportr_vo()` utility functions that outputs to console options and arguments that a user can manipulate.

### PR DESCRIPTION
**Pull request by vedhav**
### Changes
Adds a helper function `xportr_options()` which can be used to get and set xportr options. People can use the function docs to see the list of xportr options in the pkgdown site.

### The scope of `{xportr}`

`{xportr}`'s scope is to enable R users to write out submission compliant `xpt` files that can be delivered to a Health Authority or to downstream validation software programs. We see labels, lengths, types, ordering and formats from a dataset specification object (SDTM and ADaM) as being our primary focus. We also see messaging and warnings to users around applying information from the specification file as a primary focus. Please make sure your Pull Request meets this **scope of {xportr}**. If your Pull Request moves beyond this scope, please get in touch with the `{xportr}` team on [slack](https://pharmaverse.slack.com/archives/C030EB2M4GM) or create an issue to discuss.

Please check off each task box as an acknowledgment that you completed the task. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

### Changes Description

_(descriptions of changes)_

### Task List

- [x] The spirit of xportr is met in your Pull Request
- [x] Place Closes #130 into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Summary of changes filled out in the above Changes Description. Can be removed or left blank if changes are minor/self-explanatory.
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Use `styler` package and functions to style files accordingly.
- [x] Updated relevant unit tests or have written new unit tests. See our [Wiki](https://github.com/atorus-research/xportr/wiki/Style-Guide-for-Unit-Tests) for conventions used in this package.
- [x] Creation/updated relevant roxygen headers and examples. See our [Wiki](https://github.com/atorus-research/xportr/wiki/Style-Guide-for-Roxygen-Headers) for conventions used in this package.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Run `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new/updated functions occur on the "Reference" page.
- [x] Update NEWS.md if the changes pertain to a user-facing function (i.e. it has an @export tag) or documentation aimed at users (rather than developers)
- [x] Make sure that the pacakge version in the NEWS.md and DESCRIPTION file is same. Don't worry about updating the version because it will be auto-updated using the `vbump.yaml` CI.
- [x] Address any updates needed for vignettes and/or templates
- [x] Link the issue Development Panel so that it closes after successful merging.
- [x] Fix merge conflicts
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: R/options.R [hunk 1]
```diff
@@ -0,0 +1,88 @@
+#' Get or set Xportr options
```
Comment (bms63) [inline R/options.R:1]:
```suggestion
#' Get or set xportr options
```

File: R/options.R [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+#'
+#' @description
+#'
+#' There are two mechanisms for working with options for xportr. One is the
```
Comment (bms63) [inline R/options.R:5]: Very helpful!!

File: R/options.R [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+#'
+#' @section Options with `options()`:
+#'
+#' \describe{
+#' \item{xportr.df_domain_name}{defaults to `"dataset"`}:
+#'  The name of the domain "name" column in dataset metadata.
+#' \item{xportr.df_label}{defaults to `"label"`}:
+#'  The column noting the dataset label in dataset metadata.
+#' \item{xportr.domain_name}{defaults to `"dataset"`}:
+#'  The name of the domain "name" column in variable metadata.
```
Comment (bms63) [inline R/options.R:23]: Is this duplicated?

**Review by bms63 — APPROVED** [review]
I just had a few minor comments.  Feel free to incorporate or not, but this one is ready to rock!!

File: R/options.R (hunk 3 above)
Comment (vedhav) [reply R/options.R:23]: They seem to be different. One is `xportr.domain_name` while the other is `xportr.df_domain_name`
I made sure to follow the same order for these options as we had in the [xportr package](https://github.com/atorus-research/xportr/blob/bcf4c88d1e4b0a900410c309993ee23c3b2111fe/R/xportr-package.R#L12-L77) docs which was not easily visible before.

File: NEWS.md [hunk 4]
```diff
@@ -4,9 +4,9 @@
 
 * `xportr_write()` now accepts `metadata` argument which can be used to set the dataset label to stay consistent with the other `xportr_*` functions. It is noteworthy that the dataset label set using the `xportr_df_label()` function will be retained during the `xportr_write()`.
 * Exporting a new dataset `dataset_spec` that contains the Dataset Specification for ADSL. (#179)
-
 * Added a check for character variable lengths up to 200 bytes in `xpt_validate()`(#91, #189).
 * File name check is moved to strict_checks condition to allow underscores in the file name. Underscores are allowed in xpt but not per FDA requirements. (#126)
+* It is now possible to get and set the xportr options using the helper function `xportr_options()`
```
[the remark this answers is not in the document]
Comment (vedhav) [reply NEWS.md:9]: Yesss 💯

File: R/options.R (hunk 3 above)
Comment (bms63) [reply R/options.R:23]: ah yes - apologies!! you are correct.

**Pull request merged**